packages/davinci-client/src/lib/collector.types.ts

```diff
@@ -61,45 +61,36 @@ export type ActionCollectorTypes =
   | 'SocialLoginCollector'
   | 'ActionCollector';
 
-export interface ActionCollectorNoUrl<T extends ActionCollectorTypes> {
-  category: 'ActionCollector';
-  error: string | null;
-  type: T;
-  id: string;
-  name: string;
-  output: {
-    key: string;
-    label: string;
-    type: string;
-  };
-}
+type ActionCollectorOutputTypes =
+  | {
+      key: string;
+      label: string;
+      type: string;
+      url: string;
+    }
+  | {
+      key: string;
+      label: string;
+      type: string;
+    };
 
-export interface ActionCollectorWithUrl<T extends ActionCollectorTypes> {
+export interface ActionCollector<T extends ActionCollectorTypes> {
   category: 'ActionCollector';
   error: string | null;
   type: T;
   id: string;
   name: string;
-  output: {
-    key: string;
-    label: string;
-    type: string;
-    url?: string | null;
-  };
+  output: ActionCollectorOutputTypes;
 }
 
-export type ActionCollector<T extends ActionCollectorTypes> =
-  | ActionCollectorNoUrl<T>
-  | ActionCollectorWithUrl<T>;
-
 export type ActionCollectors =
-  | ActionCollectorWithUrl<'SocialLoginCollector'>
-  | ActionCollectorNoUrl<'ActionCollector'>
-  | ActionCollectorNoUrl<'FlowCollector'>
-  | ActionCollectorNoUrl<'SubmitCollector'>;
+  | ActionCollector<'SocialLoginCollector'>
+  | ActionCollector<'ActionCollector'>
+  | ActionCollector<'FlowCollector'>
+  | ActionCollector<'SubmitCollector'>;
```

Comment:
Let's make sure this and the associated types get reverted back to the previous state.